flexget/components/managed_lists/lists/ombi_list.py

```diff
@@ -0,0 +1,1038 @@
+        """Create a Ombi Entry from an TMDB ID."""
+
+        headers = request.create_json_headers()
+
+        endpoint = f"/api/v2/Search/{cls.entry_type}/{tmdb_id}"
+
+        try:
+            data = request.get(endpoint, headers=headers)
+
+            return OmbiMovie(request, data)
+        except (HTTPError, ApiError) as e:
+            log.error(f"Failed to get Ombi movie by tmdb_id: {tmdb_id}")
+            log.debug(e)
+            return None
+
+    @classmethod
+    def from_id(cls, request: OmbiRequest, entry: Entry):
+        """Create a Ombi Entry from an OMBI ID."""
+
+        if entry.get('tmdb_id'):
+            return cls.from_tmdb_id(request, entry['tmdb_id'])
+
+        if entry.get('imdb_id'):
+            return cls.from_imdb_id(request, entry['imdb_id'])
+
+        raise plugin.PluginError(
+            f"Error: Unable to find required ID to lookup Ombi {cls.entry_type}."
+        )
+
+
+class OmbiTv(OmbiEntry):
+    """Manage a TV entry in Ombi."""
+
+    entry_type = 'tv'
+
+    def __init__(
+        self,
+        request: OmbiRequest,
+        data: dict[str, Any],
+        sub_type: Literal['show', 'season', 'episode'],
+    ) -> None:
+        super().__init__(request, self.entry_type, data)
+
+        # sub_type can be show, season or episode
+        self.sub_type = sub_type
+
+    def mark_requested(self):
+        """Mark an entry in Ombi as being requested."""
+
+        api_endpoint = "api/v2/Requests/TV"
+
+        payload = {"theMovieDbId": self.data["id"]}
+
+        if self.sub_type == 'shows':
+            payload['requestAll'] = True
+        if self.sub_type == 'seasons':
+            payload['seasons'] = [
+                season
+                for season in self.data['seasonRequests']
+                if season['seasonNumber'] == self.data['tmdb_season']
+            ]
+        if self.sub_type == 'episodes':
+            payload['seasons'] = [
+                {
+                    "seasonNumber": self.data['tmdb_season'],
+                    "episodes": [{"episodeNumber": self.data['tmdb_episode']}],
+                }
+            ]
+
+        return super().mark_requested(api_endpoint, payload)
+
+    @classmethod
+    def from_tmdb_id(
+        cls, request: OmbiRequest, entry: Entry, sub_type: Literal['show', 'season', 'episode']
+    ):
+        """Create a Ombi Entry from an TVDB ID."""
+
+        headers = request.create_json_headers()
+
+        if not entry.get('tmdb_id'):
+            return None
+
+        tmdb_id = entry['tmdb_id']
+
+        endpoint = f"/api/v2/Search/{cls.entry_type}/moviedb/{tmdb_id}"
+
+        try:
+            data = request.get(endpoint, headers=headers)
+
+            # need to merge entry and data
+            entry.update(data)
+
+            return OmbiTv(request, entry, sub_type)
+        except (HTTPError, ApiError) as e:
+            log.error(f"Failed to get Ombi movie by tmdb_id: {tmdb_id}")
+            log.debug(e)
+            return None
```

Comment:
I think this PR is ready to be reviewed and then merged. But there is one point that I am unhappy with. The Ombi API REQUIRES a `tmdb_id` to request a TV show. I'm pretty sure the flexget `tmdb_lookup` does not support Tv shows. 

Does anyone have any recommendations? I feel like I only see the following paths forward.

1. We only support the `tmdb_id` and users wont be able to put `Entry`'s from Flexget into Ombi until a lookup plugin exists that will add the `tmdb_id` field to the entry.
2. We support `imdb_id` and `tvdb_id` but we have to convert it to `tmdb_id` manually. I'm not sure how. Maybe we can invoke the [tmdb_lookup](https://flexget.com/Plugins/tmdb_lookup) for them from inside ombi_list?

Edit: I also found this https://developer.themoviedb.org/reference/find-by-id which I'm guessing is how one or all the lookup_plugins work? Is there a way I can use Flexgets API key to make a request to that endpoint to lookup the `tmdb_id` using the `imdb_id`?

Reply:
I believe trakt_lookup supports tmdb_ids for shows. It would be nice if we updated tmdb_lookup to support shows as well. Is this a blocker for this PR, or things work okay without it?

Reply:
> I believe trakt_lookup supports tmdb_ids for shows. It would be nice if we updated tmdb_lookup to support shows as well. Is this a blocker for this PR, or things work okay without it?

I have been sick / busy so I haven't tested it much but at the very least it functioned as well or better than my original plugin, I tested a few of the new functions and they worked.

I think merging it will be better than leaving it as a PR. For those of us that use it we can always submit more fixes later.